src/v3/src/transformer/layout/oktaVerify/transformOktaVerifyDeviceChallengePoll.ts

```diff
@@ -12,60 +12,158 @@
     : transaction.nextStep?.relatesTo?.value?.contextualData?.challenge?.value;
 
-  uischema.elements.unshift({
+  const { challengeMethod } = deviceChallengePayload;
+  const titleElement: TitleElement = {
     type: 'Title',
-    options: { content: loc('customUri.title', 'login') },
-  } as TitleElement);
+    options: {
+      content: getTitleText(challengeMethod),
+    },
+  };
 
-  uischema.elements.push({
+  const descriptionElement: DescriptionElement = {
     type: 'Description',
     contentType: 'subtitle',
-    options: { content: loc('customUri.required.content.prompt', 'login') },
-  } as DescriptionElement);
+    options: {
+      content: getDescriptionText(challengeMethod),
+    },
+  };
 
-  uischema.elements.push({
+  const spinnerElement = {
+    type: 'Spinner',
+  } as SpinnerElement;
+
+  const openOktaVerifyButton: OpenOktaVerifyFPButtonElement = {
     type: 'OpenOktaVerifyFPButton',
     options: {
       step: nextStep.name,
       href: deviceChallengePayload.href,
+      challengeMethod,
     },
-  } as OpenOktaVerifyFPButtonElement);
+  };
 
-  uischema.elements.push({
-    type: 'Description',
-    contentType: 'subtitle',
-    options: { content: loc('customUri.required.content.download.title', 'login') },
-  } as DescriptionElement);
-
-  uischema.elements.push({
+  const cancelStep = transaction.nextStep?.name === IDX_STEP.DEVICE_CHALLENGE_POLL
+    ? 'authenticatorChallenge-cancel' : 'currentAuthenticator-cancel';
+  const cancelLink: LinkElement = {
     type: 'Link',
+    contentType: 'footer',
     options: {
-      label: loc('customUri.required.content.download.linkText', 'login'),
-      href: deviceChallengePayload.downloadHref,
+      label: loc('goback', 'login'),
+      isActionStep: true,
+      step: cancelStep,
+      actionParams: {
+        reason: 'USER_CANCELED',
+        statusCode: null,
+      },
     },
-  } as LinkElement);
+  };
+
+  uischema.elements.unshift(titleElement);
+
+  // if the previous step was identify and the challenge method is APP_LINK,
+  // we delay displaying the content because of a cold start issue with Okta Verify
+  if (challengeMethod === CHALLENGE_METHOD.APP_LINK
+      && prevTransaction?.nextStep?.name === IDX_STEP.IDENTIFY) {
+    const autoStepperElement: AutoStepperElement = {
+      type: 'AutoStepper',
+      options: {
+        nextStepIndex: 1,
+        time: FASTPASS_FALLBACK_SPINNER_TIMEOUT,
+      },
+    };
+
+    uischema.elements.push({
+      type: UISchemaLayoutType.STEPPER,
+      elements: [
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            autoStepperElement,
+            spinnerElement,
+            cancelLink,
+          ],
+        } as UISchemaLayout,
+        {
+          type: UISchemaLayoutType.VERTICAL,
+          elements: [
+            descriptionElement,
+            openOktaVerifyButton,
+            cancelLink,
+          ],
+        } as UISchemaLayout],
```

Comment:
Can you add the `viewIndex` properties to these arrays, similar to https://github.com/okta/okta-signin-widget/blob/0.1/src/v3/src/transformer/layout/googleAuthenticator/transformGoogleAuthenticatorEnroll.ts#L105

Reply:
Okay will do.  What is that used for?

Reply:
Its is used by one of the generic transformers to distinguish between the different views of a Stepper. When it traverses the layout, before this change, there is no way for it to know what elements belong to which view.

Reply:
ah awesome got it thanks!